### PR TITLE
Accept also None or Objects as params

### LINKED ISSUE #107: Support params as Object
This lib is accepting only arrays as params:

https://github.com/apoelstra/rust-jsonrpc/blob/b8c3f0d4d78440a286dd1fa7e97d6f78520b16b6/src/lib.rs#L73

I think because that was the only way available to communicate with core.
However, now there are named parameters and to support those this library should supportalso json objects as params.

Indeed jsonrpc spec https://www.jsonrpc.org/specification#request_object allows it to be an array or an object

### PR DESCRIPTION
**Pull request by RCasatta**
according to the jsonrpc spec params can be missing, or an object too.

close #107 

Breaking change

### CODE REVIEW

**Comment by apoelstra** [comment]
concept ACK. But looks like certain feature combos are broken (see CI)

**Comment by RCasatta** [comment]
I thought after the first approval the other times the CI would start automatically, but it's not the case, I hope to have "blindly" fixed it.

**Comment by RCasatta** [comment]
I think to have tested all the cases launched with the CI locally, sorry for the back and forth

**Comment by apoelstra** [comment]
>I thought after the first approval the other times the CI would start automatically, but it's not the case, I hope to have "blindly" fixed it.

All good. I blame Github, not you :).

Hit CI. Let's see how it goes.

**Review by apoelstra — APPROVED** [review]
ACK 2c3f4622a5fe7df7b434d163267d4cb21d246cdd

**Comment by apoelstra** [comment]
@RCasatta can you add a unit test that demonstrates how to provide a list, and maybe a CHANGELOG entry explaining this? This will be a breaking change which will be particularly annoying because the `RawValue` API is so annoying.

**Comment by RCasatta** [comment]
I had to force push because there were a couple of reference of `&[]` not changed and not tested in CI (the one in the Readme and the one in `integration_test/src/main.rs`)

Moreover, I added:
* the unit test commit
* a call `getblockhash` in the integration test with both list and named configuration (not launched in CI though)
* update the changelog

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,8 @@
+# 0.17.0 - 2023-06-29
```
Comment (tcharding) [inline CHANGELOG.md:1]: nit: Perhaps use "unreleased" so we don't forget to update the header when doing the release.

```suggestion
# unreleased
```

**Review by tcharding — APPROVED** [review]
ACK  178ee93f487157da3efec5f9632ca8fb5de6d9f9

**Comment by apoelstra** [comment]
Looks like some clippy lints are causing CI to fail.

**Comment by RCasatta** [comment]
Sorry again, should be fixed

**Review by tcharding — APPROVED** [review]
ACK be69ad162d76fdecfb0e3b940aae18f031ff3a83

**Comment by apoelstra** [comment]
Previously you had to provide a list with each thing wrapped in `arg()`, which was bad enough ... but now it looks like you need to wrap your lists in `Some(to_raw_value(&json!(actual_thing)).unwrap())` which is way worse.

I think we need a better API here.

**Comment by RCasatta** [comment]
You don't need the `json!` call when you have a `struct` with `serde::Serialize`

**Comment by apoelstra** [comment]
@RCasatta ah, I gotcha. So in your `list` unit test, where you write `json!([0])` it's actually fine to just do `to_raw_value(&[0])` (I tested this and it worked).

Ok, I'm happy then.

**Review by apoelstra — APPROVED** [review]
ACK 2c3f4622a5fe7df7b434d163267d4cb21d246cdd

**Pull request merged**